lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java

```diff
@@ -684,6 +694,237 @@ public void resume_publish_queue() {
+                message.action == ProtocolMessage.Action.nack);
+            final int messageCount = 6; // make 3 pending 3 queued
+            CompletionSet senderCompletion = new CompletionSet();
+            for (int i = 0; i < messageCount/2; i++) {
+                senderChannel.publish("pending_queued_message_" + i, "Test pending queued messages " + i,
+                    senderCompletion.add());
+                try { Thread.sleep(delay); } catch(InterruptedException e){}
+            }
+            assertEquals(sender.connection.connectionManager.getPendingMessages().size(),messageCount/2);
+
+            final String connectionId = sender.connection.id;
+            //now let's disconnect
+            sender.connection.connectionManager.requestState(ConnectionState.disconnected);
+            (new ConnectionWaiter(sender.connection)).waitFor(ConnectionState.disconnected);
+
+            //send some more messages while disconnected
+            for (int i = 0; i < messageCount/2; i++) {
+                senderChannel.publish("queued_message_" + i, "Test pending queued messages " + i,
+                    senderCompletion.add());
+                try { Thread.sleep(delay); } catch(InterruptedException e){}
+            }
+            //now let's unblock the ack nacks and reconnect
+            mockWebsocketFactory.blockReceive(message -> false);
+
+            /* reconnect the sender */
+            sender.connection.connect();
+            (new ConnectionWaiter(sender.connection)).waitFor(ConnectionState.connected);
+            assertEquals("Connection must be connected", ConnectionState.connected, sender.connection.state);
+            //make sure connection id is a resume success
+            assertEquals("Connection id has changed", connectionId, sender.connection.id);
+
+            /* wait for the publish callback to be called.*/
+            ErrorInfo[] errors = senderCompletion.waitFor();
+            assertTrue(
+                "Second round of messages (queued) has errors",
+                errors.length == 0
+            );
+
+            messageWaiter.waitFor(messageCount);
+
+            List<Message> received = messageWaiter.receivedMessages;
+
+            //we need to discard the first 3 - that were sent before disconnection
+            received.remove(0);
+            received.remove(1);
+            received.remove(2);
+            assertEquals(
+                "Did not receive the entire second round of messages (queued)",
+                received.size(), messageCount
+            );
+
+            //make sure that pending queue is cleared
+            assertEquals("Thre are still pending messages in the queue",
+                sender.connection.connectionManager.getPendingMessages().size(),
+                0);
+
+        } catch (AblyException e) {
+            fail("init0: Unexpected exception instantiating library");
```

Comment:
I know that this particular error message is in lots of other tests, but it's actually a copy/paste error. It could be something more meaningful for this test.

Reply:
54eae132f07d5db417ed3760c08db39f819cefb0